### ISSUE TITLE
Please make it possible to perform minimal incremental changes when synchronizing

### ISSUE THREAD

**Issue by akorn** [lsc-project/lsc#38]
Currently, when writing to an LDAP backend, lsc optimizes for efficiency; it sends bundles several changes into a single `modify` request.

Unfortunately, when many changes are made and Active Directory refuses to make them, it's often not clear which change caused the problem and why.

To help with troubleshooting, it should be possible to have lsc commit each change in a separate transaction. For example, instead of replacing the member list of a group in one go, it could delete unwanted members one by one, then add new ones one by one.

**Comment by coudot** [comment]
That would introduce a big loss of performance. Maybe a specific debug mode could be created for this.

**Comment by coudot** [comment]
We will not change this behavior in LSC. If you think this is important, you can propose a pull request and we'll look into it.

**Comment by akorn** [comment]
Ultimately we couldn't get LSC to work for synchronizing LDAP and AD and ended up writing our own tool to do it.

**Comment by coudot** [comment]
Thanks for the feedback, sorry to hear that you did not succeed to configure LSC. We already did a lot of LDAP->AD connectors, there should be nothing missing in LSC to achieve this.

**Comment by akorn** [comment]
Well, it did "work" (fsvo), but troubleshooting what went wrong when AD refused an operation was next to impossible, and implementing all the mappings and mangling we needed made the configuration too complicated.

I also found the approach taken by LSC to be very counter-intuitive (with e.g. the "pivot attribute", when there was an easily definable 1:1 mapping between AD and LDAP DNs). (I filed #39 about this, but you disagreed, which just shows how different our mindsets are -- LSC just wasn't a good match for us.)
